File: 统计二叉树中好节点的数目.cpp
class Solution {
public:
    int ans = 0;

    void dfs(TreeNode* root, int mx) {
        if (root == NULL) {
            return;
        }
        if (root->val >= mx) {
            ++ans;
            mx = root->val;
        }
        dfs(root->left, mx);
        dfs(root->right, mx);
    }
    
    int goodNodes(TreeNode* root) {
        if (root == NULL) {
            return 0;
        }
        dfs(root, root->val);
        return ans;
    }
};

File: 统计二叉树中好节点的数目.py
class Solution:
    def goodNodes(self, root: Optional[TreeNode]) -> int:
        ans = 0
        def dfs(node, mx):
            if node is None:
                return
            if node.val >= mx:
                nonlocal ans
                ans += 1
                mx = node.val
            dfs(node.left, mx)
            dfs(node.right, mx)

        dfs(root, root.val)
        return ans
